### PR TITLE
feat: add `mkinitcpio` spec

### PR DESCRIPTION
**Pull request by SebDanielsson**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Feature

**What is the current behavior? (You can also link to an open issue here)**
Autocompletion for mkinitcpio is currently not available. Closes https://github.com/withfig/autocomplete/issues/1409

**What is the new behavior (if this is a feature change)?**
Adds autocompletion for mkinitcpio

**Additional info:**

### CODE REVIEW

File: src/mkinitcpio.ts [hunk 1]
```diff
@@ -0,0 +1,179 @@
+const completionSpec: Fig.Spec = {
+  name: "mkinitcpio",
+  description: "",
```
Comment (SeparateRecords) [inline src/mkinitcpio.ts:3]: Can you add a description here?

**Review by SeparateRecords — CHANGES_REQUESTED** [review]
Looks good! One quick change then it's ready to merge 🙂

**Comment by grant0417** [comment]
Here is a generator for `-p` 

```ts
        generators: {
          script: "ls /etc/mkinitcpio.d",
          postProcess: (out) =>
            out
              .trim()
              .split("\n")
              .map((line) => ({
                name: line.replace(/\.preset$/, ""),
              })),
        },
```

**Comment by SeparateRecords** [comment]
Ah, good - let's add this generator first _then_ we'll merge 😄

**Review by SeparateRecords — APPROVED** [review]
LGTM! Thank you @SebDanielsson !

**Pull request merged**